tofu/stack: subdomain_separator variable missing — s3-snapshot resolves SSH host with default '.' separator and DNS-fails on flat-subdomain tenants

## Symptom

\`teardown.yml\` step \`Snapshot persistent state to R2 (atomic, pre-destroy)\` fails on a flat-subdomain (\`SUBDOMAIN_SEPARATOR='-'\`) fork that has a running server with valid Tofu state:

\`\`\`
s3-snapshot: SSH did not become ready after 15 attempts:
  dial tcp: lookup ssh.<user>.<base-domain> on 127.0.0.53:53: no such host
Connection closed by UNKNOWN port 65535
##[error]Process completed with exit code 2.
\`\`\`

The lookup is for \`ssh.<user>.<base-domain>\` — dot-form — but the actual DNS record is at the flat form \`ssh-<user>.<base-domain>\` because the stack is on the multi-tenant codepath added by #540.

## Root cause

\`src/nexus_deploy/pipeline.py\` at line 330 (and 631) does the right thing:

\`\`\`python
ssh_host_dns = service_host("ssh", tfvars_config.domain, tfvars_config.subdomain_separator)
\`\`\`

\`service_host\` correctly returns \`ssh-<user>.<base>\` when \`subdomain_separator='-'\`. The bug is that \`tfvars_config.subdomain_separator\` is reading as \`'.'\` (the default) instead of \`'-'\` — because the variable isn't declared on the stack side.

Specifically:

- \`tofu/control-plane/variables.tf\` (v0.58.2): **declares** \`variable \"subdomain_separator\"\`. ✓
- \`tofu/stack/variables.tf\` (v0.58.2): does **not** declare it. ✗

When \`spin-up.yml\`'s "Generate config.tfvars" step writes \`subdomain_separator = \"-\"\` into \`tofu/stack/config.tfvars\`:

\`\`\`yaml
cat > tofu/stack/config.tfvars << EOF
...
subdomain_separator  = "${{ secrets.SUBDOMAIN_SEPARATOR || '.' }}"
...

--- a/.github/workflows/destroy-all.yml
+++ b/.github/workflows/destroy-all.yml
@@ -111,12 +111,13 @@ jobs:
           server_image    = "ubuntu-24.04"
           ssh_public_key_path  = "~/.ssh/id_ed25519.pub"
           ssh_private_key_path = "~/.ssh/id_ed25519"
-          domain         = "${{ secrets.DOMAIN }}"
-          admin_email    = "${{ secrets.TF_VAR_admin_email }}"
-          user_email     = "${{ secrets.TF_VAR_user_email }}"
-          admin_username = "admin"
-          github_owner   = "${{ github.repository_owner }}"
-          github_repo    = "${{ github.event.repository.name }}"
+          domain              = "${{ secrets.DOMAIN }}"
+          subdomain_separator = "${{ secrets.SUBDOMAIN_SEPARATOR || '.' }}"
+          admin_email         = "${{ secrets.TF_VAR_admin_email }}"
+          user_email          = "${{ secrets.TF_VAR_user_email }}"
+          admin_username      = "admin"
+          github_owner        = "${{ github.repository_owner }}"
+          github_repo         = "${{ github.event.repository.name }}"
           EOF
 
           # Read services from D1 (single source of truth)

--- a/.github/workflows/teardown.yml
+++ b/.github/workflows/teardown.yml
@@ -160,12 +160,13 @@ jobs:
           server_image    = "ubuntu-24.04"
           ssh_public_key_path  = "~/.ssh/id_ed25519.pub"
           ssh_private_key_path = "~/.ssh/id_ed25519"
-          domain         = "${{ secrets.DOMAIN }}"
-          admin_email    = "${{ secrets.TF_VAR_admin_email }}"
-          user_email     = "${{ secrets.TF_VAR_user_email }}"
-          admin_username = "admin"
-          github_owner   = "${{ github.repository_owner }}"
-          github_repo    = "${{ github.event.repository.name }}"
+          domain              = "${{ secrets.DOMAIN }}"
+          subdomain_separator = "${{ secrets.SUBDOMAIN_SEPARATOR || '.' }}"
+          admin_email         = "${{ secrets.TF_VAR_admin_email }}"
+          user_email          = "${{ secrets.TF_VAR_user_email }}"
+          admin_username      = "admin"
+          github_owner        = "${{ github.repository_owner }}"
+          github_repo         = "${{ github.event.repository.name }}"
           EOF
           sed -i 's/^          //' tofu/stack/config.tfvars
 

--- a/tofu/stack/variables.tf
+++ b/tofu/stack/variables.tf
@@ -116,6 +116,24 @@ variable "domain" {
   }
 }
 
+# Mirror of tofu/control-plane/variables.tf:subdomain_separator. Issue
+# #567: stack-side workflows write this var into config.tfvars but the
+# stack variables file didn't declare it, so OpenTofu emitted a
+# "Value for undeclared variable" warning on every spin-up / teardown.
+# No stack-side resource references var.subdomain_separator today —
+# the Python pipeline reads it from config.tfvars directly to build
+# the SSH host DNS name. Declaring it here keeps the stack tfvars
+# surface in sync with control-plane and silences the warning.
+variable "subdomain_separator" {
+  description = "Separator between service subdomain and base domain. '.' for standard dot-subdomains (default, requires wildcard cert at 3rd level), '-' for flat subdomains used when provisioning tenants under a shared base domain."
+  type        = string
+  default     = "."
+  validation {
+    condition     = contains([".", "-"], var.subdomain_separator)
+    error_message = "subdomain_separator must be '.' or '-'."
+  }
+}
+
 variable "admin_email" {
   description = "Admin email for Cloudflare Access (full access including SSH)"
   type        = string

